Les_7/main.js

```diff
@@ -0,0 +1,172 @@
+const gallery = {
+    settings: {
+        galleryMain: '.my',
+        previewSelector: '.products',
+        modalImageContainer: 'gallery__modal',
+        modalImageClass: 'gallery__image',
+        modalImageScreen: 'gallery__back',
+        modalImageClose: 'gallery__close',
+        modalImageCloseSrc: 'img/close.png'
+    },
+
+    init(userSettings = {}) {
+        Object.assign(this.settings, userSettings);
+
+        const galleryContainer = document.querySelector(this.settings.previewSelector);
+        galleryContainer.addEventListener('click', (event) => this.containerClickHandler(event));
+    },
+
+    containerClickHandler(event) {
+        if (event.target.tagName !== 'IMG') {
+            return;
+        }
+        this.createGalleryModal(event.target);
+    },
+
+    createGalleryModal(img) {
+        const galleryModal = document.createElement('div');
+        galleryModal.classList.add(this.settings.modalImageContainer);
+
+        const galleryScreen = document.createElement('div');
+        galleryScreen.classList.add(this.settings.modalImageScreen);
+        galleryModal.appendChild(galleryScreen);
+
+        const galleryClose = new Image();
+        galleryClose.classList.add(this.settings.modalImageClose);
+        galleryClose.src = this.settings.modalImageCloseSrc;
+        galleryClose.addEventListener('click', (event) => {
+            this.close(event.target);
+        })
+        galleryModal.appendChild(galleryClose);
+
+        const galleryImage = new Image();
+        galleryImage.classList.add(this.settings.modalImageClass);
+        galleryImage.src = img.dataset.fullImageUrl;
+        galleryModal.appendChild(galleryImage);
+
+        document.querySelector(this.settings.galleryMain).appendChild(galleryModal);
+    },
+
+    close(closeImg) {
+        closeImg.parentElement.remove();
+    }
+}
+
+window.addEventListener('load', () => {
+    gallery.init({
+        galleryMain: '.main__content',
+    });
+})
+
+let prods = {
+    items: [
+        {
+            name: 'LV',
+            price: 27000,
+        },
+        {
+            name: 'LC',
+            price: 34000,
+        },
+        {
+            name: 'DG',
+            price: 13999,
+        }
+    ]
+}
+
+const toBuy = {
+    init() {
+        const buyClick = document.querySelector('.products');
+        buyClick.addEventListener('click', (event) => {
+            this.clickToBuy(event);
+        });
```

Comment:
if (buyClick) {
        buyClick.addEventListener('click', (event) => {
            this.clickToBuy(event);
        });
    }